code/modules/fluids/fluid_pipe_machinery.dm

```diff
@@ -0,0 +1,159 @@
+	src.network = null
+	src.initialize()
+
+/obj/machinery/fluidmachinery/unary/drain
+	var/drain_min = 0
+	var/drain_max = 0
+
+/obj/machinery/fluidmachinery/unary/drain/proc/drain()
+	if(!src.network)
+		return
+	var/turf/simulated/T = get_turf(src)
+	if(T.active_liquid?.group?.amt_per_tile && T.active_liquid.group.drain(T.active_liquid, min(rand(src.drain_min, src.drain_max), src.network.reagents.maximum_volume - src.network.reagents.total_volume)/T.active_liquid.group.amt_per_tile, src.network))
+		playsound(T, 'sound/misc/drain_glug.ogg', 50, TRUE)
+
+/obj/machinery/fluidmachinery/unary/drain/passive
+	name = "drain"
+	desc = "A drainage pipe embedded in the floor to prevent flooding. Where does the drain go? Into that pipe obviously."
+	icon_state = "drain"
+	drain_min = 2
+	drain_max = 7
+
+/obj/machinery/fluidmachinery/unary/drain/passive/process()
+	src.drain()
+
+/obj/machinery/fluidmachinery/unary/drain/passive/big
+	icon_state = "bigdrain"
+	drain_min = 6
+	drain_max = 14
+
+/obj/machinery/fluidmachinery/unary/outlet_pump
+	name = "Outlet Pump"
+	icon_state = "inlet_off"
+
+	var/on = FALSE
+	var/pullrate = 200
+
+	attack_hand(mob/user)
+		if(src.on)
+			src.on = 0
+			src.icon_state = "inlet_off"
+			boutput(user, "You turn off the fluid floor pump.")
+		else
+			src.on = 1
+			src.icon_state = "inlet_on"
+			boutput(user, "You turn on the fluid floor pump.")
+
+	process()
+		if(!src.on)
+			return
+		if(!src.network)
+			src.on = FALSE
+			src.icon_state = "inlet_off"
+			return
+		var/turf/simulated/T = get_turf(src)
+		var/datum/reagents/fluid = src.pull_from_network(src.network, src.pullrate)
+		fluid?.trans_to(T, fluid.total_volume)
+
+/obj/machinery/fluidmachinery/binary
+	var/datum/flow_network/network1
+	var/datum/flow_network/network2
+
+/obj/machinery/fluidmachinery/binary/New()
+	src.initialize_directions = src.dir|turn(src.dir, 180)
+	..()
+
+/obj/machinery/fluidmachinery/binary/disposing()
+	src.network1?.machines -= src
+	src.network2?.machines -= src
+	..()
+
+/obj/machinery/fluidmachinery/binary/initialize()
+	for(var/obj/fluid_pipe/target in get_step(src, turn(src.dir, 180)))
+		if(target.initialize_directions & get_dir(target,src))
+			src.network1 = target.network
+			src.network1.machines += src
+			break
+
+	for(var/obj/fluid_pipe/target in get_step(src, src.dir))
+		if(target.initialize_directions & get_dir(target,src))
+			src.network2 = target.network
+			src.network2.machines += src
+			break
+
+/obj/machinery/fluidmachinery/binary/refresh_network(var/datum/flow_network/network)
+	src.network1?.machines -= src
+	src.network2?.machines -= src
+	src.network1 = null
+	src.network2 = null
+	src.initialize()
+
+/obj/machinery/fluidmachinery/binary/pump
+	name = "Fluid Pump"
+	icon_state = "pump_off"
+	var/on = FALSE
+	var/pumprate = 200
+
+/obj/machinery/fluidmachinery/binary/pump/attack_hand(mob/user)
+	if(src.on)
+		src.on = 0
+		src.icon_state = "pump_off"
+		boutput(user, "You turn off the fluid pump.")
```

Comment:
i don't like this, should be a visible & possible direct user message, you know that dual proc